src/os/bluestore/BlueStore.cc

```diff
@@ -1521,7 +1523,10 @@ void BlueStore::ExtentMap::reshard(Onode *o)
       dout(20) << __func__ << " old shard end 0x" << std::hex << shard_end
 	       << std::dec << dendl;
     }
-    if (estimate && estimate + extent_avg > target) {
+    // disfavor shard boundaries that span a blob
+    bool would_span = (ep->logical_offset < max_blob_end) || ep->blob_offset;
```

Comment:
blob_offset means this Extent points to the interior of the blob, not the left-hand side.  That means that if we put a shard boundary starting at ep->logical_offset, the blob would span the boundary.